### ISSUE TITLE
Issue using turborepo and @expo/next-adapter

### ISSUE THREAD

**Issue by reggi** [expo/expo-cli#4066]
### Summary

The issue is that when using a monorepo such as turborepo or any yarn workspace, for some reason this next-js plugin and it's webpack plugins are giving me an error. This only happens however, when the project is not at the root and webpack is being hoisted outside of the nextjs project package directory. There must be a piece of code specific to looking into `./pages` directory even though i'm passing in ` { projectRoot: __dirname }`. I have tested this and gotten it working when the modules are not hoisted out of the directory via yarn workspace, and it has worked for me. It's only failing within the monorepo framework.

### Environment

Expo CLI 4.13.0 environment info:
    System:
      OS: macOS 11.4
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 17.1.0 - ~/.nvm/versions/node/v17.1.0/bin/node
      Yarn: 1.22.17 - ~/.nvm/versions/node/v17.1.0/bin/yarn
      npm: 8.1.2 - ~/.nvm/versions/node/v17.1.0/bin/npm
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.0.1, iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0
    IDEs:
      Xcode: 13.1/13A1030d - /usr/bin/xcodebuild
    npmGlobalPackages:
      expo-cli: 4.13.0
    Expo Workflow: managed

### Please specify your device/emulator/simulator platform, model and version

n/a

### Error output

ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Disabled SWC as replacement for Babel because of custom Babel configuration "babel.config.js" https://nextjs.org/docs/messages/swc-disabled
info  - automatically enabled Fast Refresh for 1 custom loader
info  - Using external babel configuration from /Users/thomasreggi/Desktop/turbo/y/apps/nextjs/babel.config.js
Warning: useTransformReactJsxExperimental has been renamed to useTransformReactJSXExperimental (capitalized JSX) in react-native@0.64.0
info  - automatically enabled Fast Refresh for 1 custom loader
error - unhandledRejection: HookWebpackError: Cannot read properties of undefined (reading 'replace')
HookWebpackError: Cannot read properties of undefined (reading 'replace')
    at makeWebpackError (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/webpack/bundle5.js:47168:9)
    at /Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/webpack/bundle5.js:31056:12
    at eval (eval at create (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/webpack/bundle5.js:141512:10), <anonymous>:12:1)
    at fn (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/webpack/bundle5.js:28479:17)
    at Hook.eval [as callAsync] (eval at create (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/webpack/bundle5.js:141512:10), <anonymous>:10:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/webpack/bundle5.js:141314:14)
    at cont (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/webpack/bundle5.js:31053:34)
    at /Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/webpack/bundle5.js:31101:10
    at symbolIterator (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/neo-async/async.js:1:14452)
    at done (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/neo-async/async.js:1:14832)
-- inner error --
TypeError: Cannot read properties of undefined (reading 'replace')
    at PagesManifestPlugin.createAssets (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/build/webpack/plugins/pages-manifest-plugin.js:35:47)
    at /Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/build/webpack/plugins/pages-manifest-plugin.js:47:22
    at fn (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/webpack/bundle5.js:28477:10)
    at Hook.eval [as callAsync] (eval at create (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/webpack/bundle5.js:141512:10), <anonymous>:10:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/webpack/bundle5.js:141314:14)
    at cont (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/webpack/bundle5.js:31053:34)
    at /Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/webpack/bundle5.js:31101:10
    at symbolIterator (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/neo-async/async.js:1:14452)
    at done (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/neo-async/async.js:1:14832)
    at /Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/neo-async/async.js:1:9438
caused by plugins in Compilation.hooks.processAssets
TypeError: Cannot read properties of undefined (reading 'replace')
    at PagesManifestPlugin.createAssets (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/build/webpack/plugins/pages-manifest-plugin.js:35:47)
    at /Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/build/webpack/plugins/pages-manifest-plugin.js:47:22
    at fn (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/webpack/bundle5.js:28477:10)
    at Hook.eval [as callAsync] (eval at create (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/webpack/bundle5.js:141512:10), <anonymous>:10:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/webpack/bundle5.js:141314:14)
    at cont (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/webpack/bundle5.js:31053:34)
    at /Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/webpack/bundle5.js:31101:10
    at symbolIterator (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/neo-async/async.js:1:14452)
    at done (/Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/neo-async/async.js:1:14832)
    at /Users/thomasreggi/Desktop/turbo/y/node_modules/next/dist/compiled/neo-async/async.js:1:9438
error - HookWebpackError: Cannot read properties of undefined (reading 'replace')

### Reproducible demo or steps to reproduce from a blank project

n/a

**Comment by brentvatne** [comment]
if you'd like to investigate this and open a pr we'd be happy to review but it's not a priority for us at the moment

**Comment by nandorojo** [comment]
Try changing your `projectRoot` to the root.

```ts
module.exports = withPlugins(
  [
    withTM,
    withFonts,
    withImages,
    withBundleAnalyzer,
    [withExpo, { projectRoot: __dirname + '/../..', jsxRuntime: 'automatic' }],
  ],
  nextConfig
)
```

PS you can use back ticks to make your code blocks easier to read: https://www.markdownguide.org/extended-syntax/#syntax-highlighting

**Comment by codinsonn** [comment]
Also encountering this issue, but only after removing all `node_modules` & then reinstalling from root with `yarn install`today, might be more of a webpack / turborepo issue instead of an Expo one?

**Comment by mhaagens** [comment]
Having the same issue with `next@12.0.7`.
The error only shows up when using `withExpo`.
If I remove it the build works.

**Comment by pleaseshutup** [comment]
minimal steps to reproduce the related bug:

```sh
# create a new nextjs site
npx create-next-app@latest
# cd to <name of app>
cd test1

# https://docs.expo.dev/guides/using-nextjs/#add-expo-to-nextjs-projects
# add expo to nextjs projects
yarn add -D @expo/next-adapter
yarn next-expo

# run next expo dev server
yarn next dev
# should crash
```

**Comment by pleaseshutup** [comment]
I lose a day of productivity on this. I went down many rabit holes.

Turns out here's all I had to do in `next.config.js`

add

webpack5: true

and where I include withTM
i removed this line I had that was really messing with it:
```
        {
            transpileModules: ['expo-next-react-navigation', 'app'],
        },
```

**Comment by mhaagens** [comment]
Next uses webpack 5 by default these days.
`withExpo` uses `withUnimodules` which needs `webpack5: true` set explicitly.

```
const webpack5 = (options.config || {}).webpack5;

const expoConfig = withUnimodules(
  config,
    {
      projectRoot: nextConfig.projectRoot || process.cwd(),
    },
    {
      supportsFontLoading: false,
      webpack5: webpack5 !== false,
    }
);
``` 

Also lost a whole day to this "gotcha".

**Comment by nandorojo** [comment]
@mhaagens the code sample you shared passes `webpack5` as `true` whenever it isn't explicitly `false`. Is that your code, or the code inside of the Next adapter?

**Comment by mhaagens** [comment]
@nandorojo Ah, yeah it does 😅 , didn't look closely enough... That's code from `withExpo.tsx` from `@expo/next-adapter`.
Either way setting `webpack5: true` solves the issue, so it must be set to false somewhere else in the process?

**Comment by rmylagary** [comment]
+1 for `webpack5: true` to fix the above issue

**Comment by marklawlor** [comment]
Hi everyone - this issue was caused by a regression in the webpack5 checking logic 

See https://github.com/expo/expo-cli/pull/4058 for more info.

The workaround is to manually set `webpack5: true` in your `next.config.js` until this is reverted.

**Comment by nandorojo** [comment]
Just encountered this issue again. It was once again fixed by adding the correct packages to `next-transpile-modules`, and then explicitly setting `webpack5: true`.
